test/unit/mem/view/src/Accessor.cpp

```diff
@@ -0,0 +1,134 @@
+/* Copyright 2021 Bernhard Manfred Gruber
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <alpaka/example/ExampleDefaultAcc.hpp>
+#include <alpaka/mem/view/Accessor.hpp>
+
+#include <catch2/catch.hpp>
+
+namespace
+{
+    constexpr auto N = 1024;
+
+    struct WriteKernelTemplate
+    {
+        template<typename Acc, typename Accessor>
+        ALPAKA_FN_ACC void operator()(Acc const& acc, Accessor data) const
+        {
+            (void) acc;
+            data[1] = 1.0f;
+            data(2) = 2.0f;
+            data[alpaka::Vec<alpaka::Dim<Acc>, alpaka::Idx<Acc>>{alpaka::Idx<Acc>{3}}] = 3.0f;
+        }
+    };
+
+    struct WriteKernelExplicit
+    {
+        template<typename Acc, typename Pointer, typename Idx>
+        ALPAKA_FN_ACC void operator()(Acc const& acc, alpaka::Accessor<Pointer, float, Idx, 1> const data) const
+        {
+            (void) acc;
+            data[1] = 1.0f;
+            data(2) = 2.0f;
+            data[alpaka::Vec<alpaka::Dim<Acc>, Idx>{Idx{3}}] = 3.0f;
+        }
+    };
+
+    struct ReadKernelTemplate
+    {
+        template<typename Acc, typename Accessor>
+        ALPAKA_FN_ACC void operator()(Acc const& acc, Accessor data) const
+        {
+            (void) acc;
+            const float v1 = data[1];
+            const float v2 = data(2);
+            const float v3 = data[alpaka::Vec<alpaka::Dim<Acc>, alpaka::Idx<Acc>>{alpaka::Idx<Acc>{3}}];
+            (void) v1;
+            (void) v2;
+            (void) v3;
+        }
+    };
+
+    struct ReadKernelExplicit
+    {
+        template<typename Acc, typename Pointer, typename Idx>
+        ALPAKA_FN_ACC void operator()(Acc const& acc, alpaka::Accessor<Pointer, const float, Idx, 1> const data) const
```

Comment:
I think this is as good as it gets:
* We have to template the `Pointer` template argument, because not every implementation uses `const float*` here (i.e. SYCL will use `sycl::global_ptr<const float>`). This will also allow us to support textures/images underneath an accessor.
* We can specify that the return value when accessing is a `const float`.
* I am still sad about the `Idx` parameter, so we can choose whether the index computations are done with e.g. `size_t` or `uint`.
* The `1` at the end gives the dimension of the accelerator.